pkg/collector/check/name_test.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package check
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNameIsExtractedWithValidExtension(t *testing.T) {
+	checkName := Name("foo.bar", ".bar")
+	assert.Equal(t, "foo", checkName)
+}
+
+func TestNameRemainsTheSameWhenAlreadyExtracted(t *testing.T) {
+	checkName := Name("foo", ".bar")
+	assert.Equal(t, "foo", checkName)
+}
+
+func TestNameCanNotBeExtractedWithWrongExtension(t *testing.T) {
+	checkName := Name("foo.d/foo.bar", ".d")
+	assert.Equal(t, "foo.d/foo.bar", checkName)
```

Comment:
it's not really obvious in the file provider code, but in this test case the check name should actually also be `foo`.

Again, it's not obvious in the provider code, but this name is actually very important in the collector logic since it also determines the name of the check that the go and python check loaders then use to load the correct check.

Reply:
Agreed, I guess this test is not really relevant though